# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/vm/FrameIter.cpp
# Commit: cda8bed19391
# Full Hash: cda8bed193914e4e6bedf875aecd9abbcc74606e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-08-02 15:37:12
# Description:
#   Bug 1910880 - Use a real JIT exit frame for Wasm JitEntry stub. r=rhunt
#   
#   JS JIT callers can call Wasm code either directly from Ion or through the JitEntry stub.
#   In the JitEntry case, we didn't have a real JS JIT exit frame. To make JS frame iteration
#   still work in this case, a special `JSJitToWasm` frame type was used to transition from Wasm
# ==============================================================================

diff -r 9b92adc2e91f -r cda8bed19391 js/src/vm/FrameIter.cpp
--- a/js/src/vm/FrameIter.cpp	Fri Aug 02 06:59:32 2024 +0000
+++ b/js/src/vm/FrameIter.cpp	Fri Aug 02 07:14:08 2024 +0000
@@ -202,15 +202,13 @@
 
     MOZ_ASSERT(wasmFrame.done());
     uint8_t* prevFP = wasmFrame.unwoundCallerFP();
-    jit::FrameType prevFrameType = wasmFrame.unwoundJitFrameType();
 
     if (mustUnwindActivation_) {
       act_->setJSExitFP(prevFP);
     }
 
     iter_.destroy();
-    iter_.construct<jit::JSJitFrameIter>(act_, prevFrameType, prevFP,
-                                         mustUnwindActivation_);
+    iter_.construct<jit::JSJitFrameIter>(act_, prevFP, mustUnwindActivation_);
     MOZ_ASSERT(!asJSJit().done());
     return;
   }